docs-src/concepts.md

```diff
@@ -292,11 +292,30 @@ TODO
 
 #### `HTTPRoute`
 
-TODO
+`HTTPRoute` is a namespace-scoped resource defined by the application
+developer.  An `HTTPRoute` is associated with an arbitrary number of
+`Gateways` and specifies a mapping between incoming HTTP requests
+matching criteria defined in the `HTTPRoute` and some target
+(typically a `Service`, but implementations may allow other resources
+as targets).
+
+TODO: Describe match rules, filters, and actions.
+
+When a `Gateway` uses an `HTTPRoute` to forward a connection, the
+`Gateway` terminates TLS.  If end-to-end encryption is desired, use
+either an `HTTPRoute` and the `Reencrypt` action or a `TCPRoute` to
+pass TLS through without decryption.
 
 #### `TCPRoute`
 
-TODO
+`TCPRoute` is a namespace-scoped resource defined by the application
+developer.  Like `HTTPRoute`, `TCPRoute` is associated with an
+arbitrary number of `Gateways`, but unlike `HTTPRoute`, `TCPRoute`
+specifies a mapping between incoming TLS connections and a target.
+`TCPRoute` does not dictate that the TLS connection encapsulate a
+specific protocol.  A common use-case for `TCPRoute` is to pass
+through connections to an HTTPS server, but in principle `TCPRoute`
+can be used to forward any protocol that is compatible with TLS.
```

Comment:
There is another use-case that needs to be addressed with TCPRoute resource:

A user wants to route traffic based on TCP ports, meaning, don't even peek into the TCP stream. This is a subset of the use-case in https://github.com/kubernetes-sigs/service-apis/issues/66.

We can merge this in and I can piggy back on top of these changes.
What I'm thinking is:
- Define a port in addition to `host` for matching
- Make the `host` optional. If it is not present, then the proxy performs port based routing, if the host is present, then the proxy performs SNI based routing.

Reply:
In this use case, is it is one-to-one: one listener to one route?  That could work.  I am wary of implicit APIs, so maybe instead of empty `host` it could be an explicit `*` value, or something more explicit?

Reply:
@hbagdi WDYT about filing a separate user story for that use case?

Reply:
Opened up https://github.com/kubernetes-sigs/service-apis/issues/96